homeassistant/components/enphase_envoy/diagnostics.py

```diff
@@ -36,11 +42,81 @@ async def async_get_config_entry_diagnostics(
     """Return diagnostics for a config entry."""
     coordinator: EnphaseUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
 
-    return async_redact_data(
-        {
-            "entry": entry.as_dict(),
-            "envoy_firmware": coordinator.envoy.firmware,
-            "data": coordinator.data,
-        },
-        TO_REDACT,
+    if TYPE_CHECKING:
+        assert coordinator.envoy.data
+    Envoy_Data: EnvoyData = coordinator.envoy.data
```

Comment:
```suggestion
    envoy_data: EnvoyData = coordinator.envoy.data
```

Reply:
done

Reply:
Is this still an issue? If not can we move ahead?